### PR TITLE
Allow for concordances in addition to reconstructions

### PR DESCRIPTION
**Pull request by mjwalter**
This allows us to present complete parts that are absent from Du Chemin's printings but present in other sources. We want them to function like reconstructions, but be listed separately under "Concordances".

### CODE REVIEW

**Comment by zolaemil** [comment]
Looks great, but would be nice to have tests too. Perhaps some of the 'DC-CanonicalMEI' test cases could be modified so they have a concordant source as well?

**Comment by mjwalter** [comment]
It sounds good to reduce code redundancy. I guess it should be possible to do this and have both in the sidebar at the same time.

**Comment by mjwalter** [comment]
Also: I think the `<rdg>` tag should have `type="reconstruction"` or `type="concordance"`, as the case may be, since it's the _reading_ that is the reconstruction or the concordance. If `<app>` gets a tag, it should be something more generic. On a practical level, we might need this because we might have a reconstruction and a concordance in the same `<app>`.

**Comment by zolaemil** [comment]
Yes, in this case it would be better if the type attribute was defined on
the rdg. On the other hand, the type attribute implicitly encodes that
these readings are of different nature: a reconstruction has a responsible
editor, a concordance has a source.

By the way, so far only variants were encoded as rdg + a source attribute.
You may want to double check if there's any code that could break now due
to an assumption that rdg+source implies a variant!

2014-08-25 13:12 GMT+01:00 Micah Walter notifications@github.com:

**Comment by mjwalter** [comment]
“On the other hand, the type attribute implicitly encodes that
these readings are of different nature: a reconstruction has a responsible
editor, a concordance has a source.”

As you say later, we still seem to need the type attribute—otherwise, the rdg element will look just like a variant. It's probably worth marking emendations and variants explicitly with the `type="emendation"` and `type="variant"`; at least, it can't hurt.

**Comment by zolaemil** [comment]
Sure, more explicit markup can only be a good thing.

2014-08-25 13:54 GMT+01:00 Micah Walter notifications@github.com:

**Comment by mjwalter** [comment]
I revised the code to reduce redundancy and increase abstraction. I also added tests for concordances, with and without reconstructions.

**Comment by zolaemil** [comment]
The first five test cases don't work because of file name inconsistencies. Perhaps some files were missed out from the commits?

The problematic test cases: 
- Sample Compact Viewer
- Single voice with one concordant source
- MEI with reconstructions, emendations and variants (1)
- MEI with reconstructions, emendations and variants (1) - Without Styling
- MEI with reconstructions, emendations and variants (2)

**Comment by mjwalter** [comment]
I was actually struggling a bit with GitHub about this. It's all right in my repository now; I'll try again in the morning.

**Comment by mjwalter** [comment]
Looking at my fork on GitHub (https://github.com/mjwalter/meiView/tree/concordance/test) I see everything as it should be. Perhaps the merge tool is not renaming files?

**Comment by zolaemil** [comment]
The file names are updated, but they point to non-existing files, for example:
https://github.com/mjwalter/meiView/blob/concordance/test/DC-CanonicalMEI.nostyle.html#L58

Do all tests work for you?

**Comment by mjwalter** [comment]
Oh I see now! I didn't update the tests.html file which linked to all the others. I've pushed the revisions.

File: js/meiview.js [hunk 1]
```diff
@@ -19,9 +19,35 @@
 * permissions and limitations under the License.
 ***/
 
+
 // the default line thickness is 2, but this renders poorly with meiView's scaling
 if (typeof Vex !== 'undefined') Vex.Flow.STAVE_LINE_THICKNESS = 1;
 
+
+// Constants associated with each variant (supplied part) type
+cons = function(var_type) {
```
Comment (zolaemil) [inline js/meiview.js]: This creates a global object that isn't a good practice, and also the name is way too generic for what it is.
For sure it must be under the meiView namespace object and I wouldn't mind it you could find a more specific/descriptive name too.

**Comment by zolaemil** [comment]
Thanks it's working now! Thanks for the supplied-parts abstractions! My last request is to avoid creating global obejcts move cons() under the meiView namespace! And perhaps give it a more specific name too, at the moment it sounds if it was a generic function...

File: js/meiview.js (hunk 1 above)
Comment (mjwalter) [inline js/meiview.js]: You're right. I'll see what else I can do.

**Comment by mjwalter** [comment]
Revised and tested. I think this way makes more logical sense :)

**Pull request merged**